### ISSUE TITLE
Can livy_load_scala_sources function load source files parallely?

### ISSUE THREAD

**Issue by Jassy1994** [rstudio/sparklyr#1865]
# Reporting an Issue with sparklyr

I use sparklyr connect spark by livy, but it costs lots of time in livy_load_scala_sources, this func will load source files and execute statement one by one，can it modified to load source parallely? I try to use parLapply replace lapply, but it takes mistake，I cannot make sure whether those scala source file has dependencies, and can not be executed parallely。
I modified livy code to support execute statement parallely，it can run in rest api，but in sparklyr it report this:
```
Error in value[[3L]](cond) : 
  Failed to initialize livy connection: 3 nodes produced errors; first error: Failed to load utils.scala: Failed to execute Livy statement with error: <console>:32: error: illegal start of simple expression
Traceback: }
```
and error in livy ui：
<img width="1235" alt="2019-01-30 17 51 49" src="https://user-images.githubusercontent.com/17550838/51974060-051fd500-24ba-11e9-98b2-fecbbd148552.png">

**Comment by javierluraschi** [comment]
This is a great suggestion that brings 5X improvements to most operations, here is a PR to be merged shortly: https://github.com/rstudio/sparklyr/pull/1868

**Comment by Jassy1994** [comment]
Hi @javierluraschi 
Thanks a lot for answering my question and I see in the PR，spark_version should be set to 2.3.0，but my spark version is 2.2.X，can spark version 2.2 support the new feature in this PR？

**Comment by javierluraschi** [comment]
@Jassy1994 Yes, previous versions are also supported, if you are using `2.2.1` just set `spark_version = "2.2.1"`.
